### PR TITLE
Use a python standard library for parsing and constructing the icon and image url

### PR DESCRIPTION
**Pull request by GuelorEmanuel**
Refactored the code by using a standard library for parsing and constructing the URL.

### CODE REVIEW

File: previews/model.py [hunk 1]
```diff
@@ -148,17 +135,7 @@ def fetch(self, timeout=1):
...
-
-                if image_link.find("http") != -1:
-                    self.image = image_link
-                elif image_link.startswith("//"):
-                    self.image = protocol + image_link
-                elif image_link.startswith("/"):
-                    self.image = base_url + image_link
-                else:
-                    self.image = self.url + image_link
+                self.image = self.get_full_url(image["content"])
```
Comment (andreirtaylor) [inline previews/model.py:138]: I dont get what this parameter is doing. Could you explain it in a comment?

Comment (GuelorEmanuel) [reply previews/model.py:138]: Are you talking about `def get_full_url(self, href):` argument? If so, the function is just expecting what we assumed to be a link/href of the icon or image and then we are using that to build our full icon url and image url by adding the parts that are missing.

For example if the passed in argument was  "/DynamicContent/Resources/Images/favicon.ico" we would rebuild it by replacing the empty `scheme` and `netloc` with that of the original `url` that was specified in the query paramater

Comment (andreirtaylor) [reply previews/model.py:138]: Why do we need to look at both the image/icon href and the url? Shouldnt we be able to get all of the information from just the URL?

Comment (GuelorEmanuel) [reply previews/model.py:138]: The image/icon href could also be an absolute URL. In that case, we don't care about the original URL thats was passed in as a query parameter. We are only getting all of the information that we need from the url:`https://previews.ada.support/?url=https://www.blackriflecoffee.com/pages/liberty-babgab` when icon/image href is not an absolute URL.

**Review by andreirtaylor — APPROVED** [review]

**Pull request merged**